### ISSUE TITLE
Unsupported files ignored even though override w/parser exists

### ISSUE THREAD

**Issue by jungaretti** [prettier/prettier#11141]
**Environments:**

- Prettier Version: 2.3.1
- Running Prettier via: CLI
- Runtime: Node.js v16.4.0
- Operating System: Linux
- Prettier plugins (if any): None

**Steps to reproduce:**

```bash
mkdir prettier-bug
cd prettier-bug

# Install Prettier
npm init -y
npm install --save-dev prettier@2.3.1
echo '{"overrides":[{"files":"*.abc","options":{"parser":"html"}}]}' >.prettierrc

# Create test file
mkdir src
echo '<!DOCTYPE html><html lang="en"><head><meta charset="UTF-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Prettier Rocks!</title></head><body><p>Prettier is the best!</p></body></html>' >src/test.abc

# Try to format src
npx prettier --write src
```

**Expected behavior:**

Since I configured an override for all `.abc` files, I expect Prettier to format `test.abc` using its `html` parser.

```bash
src/test.abc 36ms
```

**Actual behavior:**

Prettier ignores `test.abc` and complains that there aren't any supported files in `src`.

```bash
[error] No supported files were found in the directory: "src".
```
If I explicitly tell Prettier to format `test.abc` with `npx prettier --write test.abc`, everything works as intended. (Prettier formats the file using its `html` parser, and all is good in the world.) I have a couple `.njk` files that I'd like to format using the `html` parser. For now, I'll use `prettier --write src src/**/*.njk`.

It seems like Prettier should recognize files that have an override w/parser instead of ignoring them.

https://prettier.io/docs/en/configuration.html#setting-the-parserdocsenoptionshtmlparser-option

**Comment by fisker** [comment]
This is duplicate of https://github.com/prettier/prettier/issues/8918, but we don't have an open issue tracking it. So , we can keep this open.

**Comment by fisker** [comment]
And now we have `--ignore-unknown` option now, maybe we can glob all files.

**Comment by YodaDaCoda** [comment]
I found that doing `prettier --write **` instead of `prettier --write .` works. It means I have to ignore a bunch more stuff in my `.prettierignore` (e.g. `*.png`) but it works.

**Comment by jungaretti** [comment]
Duplicate of https://github.com/prettier/prettier/issues/8924

**Comment by tonyhallett** [comment]
Seems like a bug rather than an enhancement.

> prettier --write .
This command formats all files supported by Prettier in the current directory and its subdirectories.

> Combined with overrides you can teach Prettier how to parse files it does not recognize.

Surely it should then support the .parcelrc file I have taught it to recognize ?
```
{
    "overrides": [
        {
            "files": ".parcelrc",
            "options": {
                "parser": "json"
            }
        }
    ]
}
```

Note that 

> For example, to get Prettier to format its own .prettierrc file, you can do:
```
{
  "overrides": [
    {
      "files": ".prettierrc",
      "options": { "parser": "json" }
    }
  ]
}
```
will be formatted with prettier --write . unlike the .parcelrc file.
In fact it will get formatted without the override.

**Comment by fehnomenal** [comment]
I have the same issue with `prettier-plugin-svelte` and running `prettier --check .` that it does not pick up `.svelte` files. Calling directly on a file `prettier --check abc.svelte` works correctly.

I guess the function responsible for this issue is `getSupportedFilesGlob`. The function runs when one of the supplied file patterns resolves to a directory and uses all loaded languages to build the glob. Apparently at this point the context does not know about overrides or plugins.

**Comment by TyIsI** [comment]
> I found that doing `prettier --write **` instead of `prettier --write .` works. It means I have to ignore a bunch more stuff in my `.prettierignore` (e.g. `*.png`) but it works.

My work-around is to execute prettier like this `yarn prettier -l '**/*'` or `yarn prettier -l 'sql/**' 'src/**'` with the following `.prettierignore` file:
```
*

!sql
!sql/**
!sql/**/*.sql

!src
!src/**
!src/**/*.ts
```

For this workaround, `--ignore-unknown` was not required.

IMHO, much like a .gitignore file, it's never a bad idea to do an inverse filter for formatters.

The above did not work with `.` or `sql src` as the file parameters.
Obviously specifically adding all the files with `$(find sql src -type f)` did yield results.

**Comment by thorn0** [comment]
Fixed by #15433
Check v3.1 please. If not fixed, let's reopen.

**Comment by jungaretti** [comment]
Yes, this is fixed in v3.1. Nice! I confirmed with a small test project: https://github.com/jungaretti/playground/tree/prettier-11141

v3.0 shows the same error as before:
<img width="456" alt="No supported files were found in the directory: src" src="https://github.com/prettier/prettier/assets/19893438/0d952ed1-bf02-4168-a548-3032ba302561">

Then, I installed v3.1 with `npm install --save-dev --save-exact prettier@3.1`

v3.1 formats `test.abc` as expected:
<img width="466" alt="src/test.abc 16ms" src="https://github.com/prettier/prettier/assets/19893438/08cc0082-1542-44b0-ac8c-5fa089167e3b">

**Comment by thorn0** [comment]
@jungaretti Thanks!
